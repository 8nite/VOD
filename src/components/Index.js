import React from 'react'
import VideoList from './VideoList'
import 'react-responsive-carousel/lib/styles/carousel.min.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import Header from './Header';

class Index extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      entries: []
    }
  }

  componentDidMount() {
    fetch(process.env.REACT_APP_NOT_SECRET_CODE + '/?filter=' + this.props.filter)
      .then(response => response.json())
      .then(entries => {
        this.setState(entries)
      });
  }

  render() {
    return (
      <div>
        <Header changeFilter={this.props.changeFilter}/>
        <VideoList entries={this.state.entries} filter={this.props.filter} watched={this.props.watched} onEnded={this.props.onEnded}/>
      </div>          
    )
  }
}

export default Index
