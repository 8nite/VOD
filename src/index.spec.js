import React from 'react';
import Index from './index'
import { shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const testindex = () => (
    <Index>
    </Index>
)

describe('index tester', () => {
    it('rendering correct loads', () => {
        const Index = shallow(testindex());
        expect(Index.find('Index').map((node) => node.text())).toHaveLength(1)
    })
    it('testing onChangeFilter function', () => {


    })
})
