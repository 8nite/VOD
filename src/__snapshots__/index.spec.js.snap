// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index tester testing onChangeFilter function 1`] = `
<div>
  <header>
    <nav
      className="navbar navbar-expand-lg navbar-dark bg-dark"
    >
      <a
        className="navbar-brand"
        href="#home"
      >
        One Page Video Web App
      </a>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          className="mr-auto navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            className="nav-link"
            data-rb-event-key={null}
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
          >
            <button
              className="btn btn-secondary"
              disabled={false}
              type="button"
              value=""
            >
              ALL
            </button>
          </a>
          <a
            className="nav-link"
            data-rb-event-key={null}
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
          >
            <button
              className="btn btn-primary"
              disabled={false}
              type="button"
              value="movies-comedy"
            >
              Comedy
            </button>
          </a>
          <a
            className="nav-link"
            data-rb-event-key={null}
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
          >
            <button
              className="btn btn-secondary"
              disabled={false}
              type="button"
              value="movies-biography"
            >
              Biography
            </button>
          </a>
          <a
            className="nav-link"
            data-rb-event-key={null}
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
          >
            <button
              className="btn btn-success"
              disabled={false}
              type="button"
              value="movies-action"
            >
              Action
            </button>
          </a>
          <a
            className="nav-link"
            data-rb-event-key={null}
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
          >
            <button
              className="btn btn-warning"
              disabled={false}
              type="button"
              value="movies-mystery"
            >
              Mystery
            </button>
          </a>
          <a
            className="nav-link"
            data-rb-event-key={null}
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
          >
            <button
              className="btn btn-danger"
              disabled={false}
              type="button"
              value="movies-horror"
            >
              Horror
            </button>
          </a>
          <a
            className="nav-link"
            data-rb-event-key={null}
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
          >
            <button
              className="btn btn-light"
              disabled={false}
              type="button"
              value="movies-drama"
            >
              Drama
            </button>
          </a>
          <a
            className="nav-link"
            data-rb-event-key={null}
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
          >
            <button
              className="btn btn-info"
              disabled={false}
              type="button"
              value="movies-crime"
            >
              Crime
            </button>
          </a>
        </div>
      </div>
    </nav>
  </header>
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <div
        className="row w-100"
      >
        <div
          className="col align-self-start"
        >
          movies-comedy
        </div>
        <div
          className="col align-self-center"
        />
      </div>
      <div
        className="row w-100"
      >
        <div
          className="col align-self-start"
        >
          movies-biography
        </div>
        <div
          className="col align-self-center"
        />
      </div>
      <div
        className="row w-100"
      >
        <div
          className="col align-self-start"
        >
          movies-action
        </div>
        <div
          className="col align-self-center"
        />
      </div>
      <div
        className="row w-100"
      >
        <div
          className="col align-self-start"
        >
          movies-mystery
        </div>
        <div
          className="col align-self-center"
        />
      </div>
      <div
        className="row w-100"
      >
        <div
          className="col align-self-start"
        >
          movies-horror
        </div>
        <div
          className="col align-self-center"
        />
      </div>
      <div
        className="row w-100"
      >
        <div
          className="col align-self-start"
        >
          movies-drama
        </div>
        <div
          className="col align-self-center"
        />
      </div>
      <div
        className="row w-100"
      >
        <div
          className="col align-self-start"
        >
          movies-crime
        </div>
        <div
          className="col align-self-center"
        />
      </div>
    </div>
  </div>
</div>
`;
